include/deal.II/base/function_parser.h

```diff
@@ -26,10 +26,7 @@
 #include <vector>
 #include <map>
 
-namespace mu
-{
-  class Parser;
-}
+#include <muParser.h>
```

Comment:
I am surprised that this worked before given that we create a `std::vector<mu::Parser>>` down below.

Would you mind wrapping this inclusion with a `#ifdef DEAL_II_WITH_MUPARSER` check?